modulus/experimental/models/sfno/sfnonet.py

```diff
@@ -0,0 +1,887 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+import torch
+import torch.nn as nn
+from torch.utils.checkpoint import checkpoint
+from torch.cuda import amp
+from dataclasses import dataclass
+from typing import Any, List
+
+from modulus.models.layers import get_activation
+
+# helpers
+from modulus.experimental.models.sfno.layers import (
+    trunc_normal_,
+    DropPath,
+    MLP,
+    EncoderDecoder,
+)
+
+# import global convolution and non-linear spectral layers
+from modulus.experimental.models.sfno.layers import (
+    SpectralAttention2d,
+    SpectralAttentionS2,
+)
+
+from modulus.experimental.models.sfno.s2convolutions import SpectralConvS2
+
+# get spectral transforms from torch_harmonics
+import torch_harmonics as th
+import torch_harmonics.distributed as thd
+
+# wrap fft, to unify interface to spectral transforms
+from modulus.experimental.models.sfno.layers import RealFFT2
+from modulus.experimental.utils.sfno.distributed.layers import (
+    DistributedRealFFT2,
+    DistributedInverseRealFFT2,
+    DistributedMLP,
+    DistributedEncoderDecoder,
+)
+from modulus.utils.sfno.distributed.mappings import (
+    scatter_to_parallel_region,
+    gather_from_parallel_region,
+)
+
+# more distributed stuff
+from modulus.utils.sfno.distributed import comm
+
+# layer normalization
+from apex.normalization import FusedLayerNorm
+from modulus.utils.sfno.distributed.layer_norm import (
+    DistributedInstanceNorm2d,
+)
+
+from modulus.models.module import Module
+from modulus.models.meta import ModelMetaData
+
+
+@dataclass
+class MetaData(ModelMetaData):
```

Comment:
Rename to something more specific like `SFNOMetaData`. Also would prefer this be moved to right above where the model is.

Reply:
Fixed this